testSuite/shared/src/test/scala/org/scalajs/testsuite/javalib/time/InstantTest.scala

```diff
@@ -0,0 +1,357 @@
+package org.scalajs.testsuite.javalib.time
+
+import java.time.{LocalDate, DateTimeException, Instant}
+import java.time.temporal.{UnsupportedTemporalTypeException, ChronoUnit, ChronoField}
+
+import org.junit.Test
+import org.junit.Assert._
+import org.scalajs.testsuite.utils.AssertThrows._
+
+/** Created by alonsodomin on 26/12/2015. */
+class InstantTest extends TemporalTest[Instant] {
+  import DateTimeTestUtil._
+  import ChronoField._
+  import ChronoUnit._
+
+  val someInstant = Instant.ofEpochMilli(928392983942L)
+  val samples = Seq(Instant.EPOCH, Instant.MIN, Instant.MAX, someInstant)
+
+  def isSupported(field: ChronoField): Boolean =
+    field == INSTANT_SECONDS || field == NANO_OF_SECOND || field == MICRO_OF_SECOND ||
+      field == MILLI_OF_SECOND
+
+  def isSupported(unit: ChronoUnit): Boolean =
+    unit.isTimeBased || unit == DAYS
+
+  @Test def getLong(): Unit = {
+    for (field <- ChronoField.values() if isSupported(field))
+      assertEquals(0L, Instant.EPOCH.getLong(field))
+
+    assertEquals(-31557014167219200L, Instant.MIN.getLong(INSTANT_SECONDS))
+    assertEquals(0L, Instant.MIN.getLong(NANO_OF_SECOND))
+    assertEquals(0L, Instant.MIN.getLong(MICRO_OF_SECOND))
+    assertEquals(0L, Instant.MIN.getLong(MILLI_OF_SECOND))
+
+    assertEquals(31557014167219200L, Instant.MAX.getLong(INSTANT_SECONDS))
+    assertEquals(0L, Instant.MAX.getLong(NANO_OF_SECOND))
+    assertEquals(0L, Instant.MAX.getLong(MICRO_OF_SECOND))
+    assertEquals(0L, Instant.MAX.getLong(MILLI_OF_SECOND))
+
+    assertEquals(928392983L, someInstant.getLong(INSTANT_SECONDS))
+    assertEquals(942000000L, someInstant.getLong(NANO_OF_SECOND))
+    assertEquals(942000L, someInstant.getLong(MICRO_OF_SECOND))
+    assertEquals(942L, someInstant.getLong(MILLI_OF_SECOND))
+  }
+
+  @Test def getEpochSecond(): Unit = {
+    assertEquals(0L, Instant.EPOCH.getEpochSecond)
+    assertEquals(-31557014167219200L, Instant.MIN.getEpochSecond)
+    assertEquals(31557014167219200L, Instant.MAX.getEpochSecond)
+    assertEquals(928392983L, someInstant.getEpochSecond)
+  }
+
+  @Test def getNano(): Unit = {
+    assertEquals(0, Instant.EPOCH.getNano)
+    assertEquals(0, Instant.MIN.getNano)
+    assertEquals(0, Instant.MAX.getNano)
```

Comment:
The spec says that `Instant.MAX` is '1000000000-12-31T23:59:59.999999999Z', therefore this should by `assertEquals(999999999, Instant.MAX.getNano)`